modules/navier_stokes/include/userobjects/RhieChowFaceFluxProvider.h

```diff
@@ -0,0 +1,47 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GeneralUserObject.h"
+#include "BlockRestrictable.h"
+#include "FaceArgInterface.h"
+#include "MooseTypes.h"
+
+class MooseMesh;
+namespace libMesh
+{
+class Elem;
+class MeshBase;
+}
+
+class RhieChowFaceFluxProvider : public GeneralUserObject,
+                                 public BlockRestrictable,
+                                 public FaceArgInterface
+{
+public:
+  static InputParameters validParams();
+  RhieChowFaceFluxProvider(const InputParameters & params);
+
+  /**
+   * Retrieve the volumetric face flux, will not include derivatives
```

Comment:
should we have a version of this that includes derivatives?
then the Real value is simply taking the raw value

Reply:
not sure we can actually. they would only differ in the return type. and C++ does not like that

Reply:
Yeah we can add it with a different name. Right now this is only used in the postprocessor which requires Real.

Reply:
I am not going to add it considering that this is used by one thing at this point and the doco states that no AD information is included. I don't want to add a `getVolumetricFaceFluxAD` function just with no use cases.